assets/js/googlesitekit/datastore/site/notifications.test.js

```diff
@@ -21,18 +21,129 @@
-		expect( Object.keys( actions ) ).toEqual(
-			expect.arrayContaining( actionsToExpect )
+	let registry;
+
+	beforeEach( () => {
+		registry = createTestRegistry();
+		const response = true;
+		fetchMock.post(
+			/^\/google-site-kit\/v1\/core\/site\/data\/mark-notification/,
+			{ body: JSON.stringify( response ), status: 200 }
 		);
 	} );
 
-	it( 'has appropriate notification selectors', () => {
-		const selectorsToExpect = [ 'getNotifications' ];
-		expect( Object.keys( selectors ) ).toEqual(
-			expect.arrayContaining( selectorsToExpect )
-		);
+	afterEach( () => {
+		unsubscribeFromAll( registry );
+	} );
+
+	describe( 'actions', () => {
+		describe( 'fetchMarkNotification', () => {
+			it.each( [
+				[ 'abc', 'accepted' ],
+				[ 'abc', 'dismissed' ],
+			] )(
+				'properly marks a notification when the notification ID is "%s" and notification state is "%s".',
+				( notificationID, notificationState ) => {
+					expect( () =>
+						registry.dispatch( CORE_SITE ).fetchMarkNotification( {
+							notificationID,
+							notificationState,
+						} )
+					).not.toThrow();
+				}
+			);
+			it.each( [
+				[ 'abc', 'test' ],
+				[ 'abc', undefined ],
+				[ undefined, 'accepted' ],
+			] )(
+				'throws an error while trying to marks a notification when the notification ID is "%s" and notification state is "%s".',
+				( notificationID, notificationState ) => {
+					expect( () =>
+						registry.dispatch( CORE_SITE ).fetchMarkNotification( {
+							notificationID,
+							notificationState,
+						} )
+					).toThrow();
+				}
+			);
+		} );
+		describe( 'acceptNotification', () => {
+			it.each( [ [ 'abc' ] ] )(
+				'properly accepts a notification when the notification ID is "%s".',
+				( notificationID ) => {
+					expect( () =>
+						registry
+							.dispatch( CORE_SITE )
+							.acceptNotification( notificationID )
+					).not.toThrow();
+				}
+			);
+			it.each( [ [ undefined, true ] ] )(
+				'throws an error when trying to accept a notification when the notification ID is "%s".',
+				( notificationID ) => {
+					expect( () =>
+						registry
+							.dispatch( CORE_SITE )
+							.acceptNotification( notificationID )
+					).toThrow();
+				}
+			);
+		} );
+		describe( 'dismissNotification', () => {
+			it.each( [ [ 'abc' ] ] )(
+				'properly dismisses a notification when the notification ID is "%s".',
+				( notificationID ) => {
+					expect( () =>
+						registry
+							.dispatch( CORE_SITE )
+							.dismissNotification( notificationID )
+					).not.toThrow();
+				}
+			);
+			it.each( [ [ undefined, true ] ] )(
+				'throws an error when trying to dismiss a notification when the notification ID is "%s".',
+				( notificationID ) => {
+					expect( () =>
+						registry
+							.dispatch( CORE_SITE )
+							.dismissNotification( notificationID )
+					).toThrow();
+				}
+			);
+		} );
+
+		it( 'has appropriate notification actions', () => {
+			const actionsToExpect = [
```

Comment:
Let's update this test to use the `.each` approach because it will give us better information about which action is missing on failure.

Reply:
https://github.com/google/site-kit-wp/pull/4349/commits/b95372ab1f66c942aae27c99218ea57492595555